libsolidity/codegen/ReturnInfo.cpp

```diff
@@ -41,7 +41,7 @@ ReturnInfo::ReturnInfo(EVMVersion const& _evmVersion, FunctionType const& _funct
 			returnTypes = _functionType.returnParameterTypesWithoutDynamicTypes();
 
 		for (auto const& retType: returnTypes)
-			if (retType->isDynamicallyEncoded())
+			if (retType->decodingType()->isDynamicallyEncoded())
```

Comment:
Please add an assertion about decoding type not being null and remove the last else case - at least in this way it does not make sense :)